homework-2_js/script.js

```diff
@@ -0,0 +1,158 @@
+//1 Запросить у пользователя его возраст и определить, кем он является: ребенком (0–2), подростком (12–18), взрослым (18_60) или пенсионером (60– ...).
+let ageOfThePerson = +prompt("Який твiй вiк?");
+if (ageOfThePerson <= 2) {
+  alert(`Ти дитина`);
+} else if (ageOfThePerson >= 12 && ageOfThePerson <= 18) {
+  alert(`Ти пiдросток`);
+} else if (ageOfThePerson >= 18 && ageOfThePerson <= 60) {
+  alert(`Ти дорослий`);
+} else if (ageOfThePerson >= 60) {
+  alert(`Ти пенсiонер`);
+}
+//2 Запросить у пользователя число от 0 до 9 и вывести ему спецсимвол, который расположен на этой клавише (1–!, 2–@, 3–# и т. д).
+let numericSpecialCharacters = +prompt("Вкажи число с 1 до 9");
+switch (numericSpecialCharacters) {
+  case 1:
+    alert(`Спецсимвол "!"`);
+    break;
+  case 2:
+    alert(`Спецсимвол "@"`);
+    break;
+  case 3:
+    alert(`Спецсимвол "#"`);
+    break;
+  case 4:
+    alert(`Спецсимвол "$"`);
+    break;
+  case 5:
+    alert(`Спецсимвол "%"`);
+    break;
+  case 6:
+    alert(`Спецсимвол "^"`);
+    break;
+  case 7:
+    alert(`Спецсимвол "&"`);
+    break;
+  case 8:
+    alert(`Спецсимвол "*"`);
+    break;
+  case 9:
+    alert(`Спецсимвол "("`);
+}
+//3 Запросить у пользователя трехзначное число и проверить, есть ли в нем одинаковые цифры.
+let number = +prompt("Введiть трьохкратне число :");
+let number1 = parseInt(number / 100);
+let number2 = parseInt(number / 10) % 10;
+let number3 = number % 10;
+if (number1 === number2 || number2 === number3 || number1 === number3)
+  alert(`Найдено совмiснiсть цифер`);
+else {
+  alert(`Совмiснiсть ненайдена`);
+}
+//4 Чи могло так бути ?
+// Запросить у пользователя год и проверить, високосный он или нет. Високосный год либо кратен 400, либо кратен 4 и при этом не кратен 100.
+let aLeapYear = +prompt(`Впишiть рiк:`);
+if (aLeapYear % 4 === 0) {
+  alert(`Цей рiк високосний`);
+} else {
+  alert(`Цей рiк не э високосним`);
+}
+//4.2 Запросить у пользователя год и проверить, високосный он или нет. Високосный год либо кратен 400, либо кратен 4 и при этом не кратен 100.
+let aLeapYear2 = +prompt(`Впишiть рiк:`);
+if (aLeapYear2 % 4 === 0 || (aLeapYear2 % 100 != 0 && aLeapYear2 % 400 === 0)) {
+  alert(`Цей рiк високосний`);
+} else {
+  alert(`Цей рiк не э високосним`);
+}
+//5 Запросить у пользователя пятиразрядное число и определить, является ли оно палиндромом.
+const number4 = +prompt(`Напишiть пятизначне число`);
+let firstDigit = Math.floor(number4 / 10000);
+let secondDigit = Math.floor(number4 / 1000) - 10 * firstDigit;
+let thirdDigit =
+  Math.floor(number4 / 100) - (100 * firstDigit + 10 * secondDigit);
+let fourthDigit =
+  Math.floor(number4 / 10) -
+  (1000 * firstDigit + 100 * secondDigit + 10 * thirdDigit);
+let fifthDigit =
+  number4 -
+  (10000 * firstDigit +
+    1000 * secondDigit +
+    100 * thirdDigit +
+    10 * fourthDigit);
+if (firstDigit === 0 || number4 < 10000 || number4 > 99999) {
```

Comment:
паліндромом є число наприклад 12321, тобто читається з двох сторін однаково
я видно в ньому однакові перша і остання цифра, і друга і передостання, що треба і перевірити
```suggestion
if (firstDigit === fifthDigit && secondDigit === fourthDigit) {
```